docs/workflows/create_workflow_block.md

```diff
@@ -0,0 +1,1898 @@
+
+* **Specifies Dimensionality:** It details the properties related to input and output dimensionality.
+
+* **Indicates Batch Inputs and Empty Values:** It informs the Execution Engine whether the step accepts batch 
+inputs and empty values.
+
+* **Ensures Compatibility:** It dictates the compatibility with different Execution Engine versions to maintain 
+stability. For more details, see [versioning](/workflows/versioning/).
+
+### Scaffolding for manifest
+
+To understand how manifests work, let's define one step-by-step. The example block that we build here will be 
+calculating images similarity. We start from imports and class scaffolding:
+
+```python
+from typing import Literal
+from inference.core.workflows.prototypes.block import (
+    WorkflowBlockManifest,
+)
+
+class ImagesSimilarityManifest(WorkflowBlockManifest):
+    type: Literal["my_plugin/images_similarity@v1"] 
+    name: str
+```
+
+This is the minimal representation of manifest. It defines two special fields that are important for 
+Compiler and Execution engine:
+
+* `type` - required to parse syntax of Workflows definitions based on dynamic pool of blocks - this is the 
+[`pydantic` type discriminator](https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions) that let
+Compiler understand which block manifest is to be verified when parsing specific step in Workflow definition
+
+* `name` - this property will be used to give the step unique name and let other steps selects it via selectors
+
+### Adding batch-oriented inputs
+
+We want our step to take two batch-oriented inputs with images to be compared - so effectively
+we will be creating SIMD block. 
+
+??? example "Adding batch-oriented inputs"
+    
+    Let's see how to add definitions of those inputs to manifest: 
+
+    ```{ .py linenums="1" hl_lines="2 6 7 8 9 17 18 19 20 21 22"}
+    from typing import Literal, Union
+    from pydantic import Field
+    from inference.core.workflows.prototypes.block import (
+        WorkflowBlockManifest,
+    )
+    from inference.core.workflows.execution_engine.entities.types import (
+        StepOutputImageSelector,
+        WorkflowImageSelector,
+    )
+    
+    class ImagesSimilarityManifest(WorkflowBlockManifest):
+        type: Literal["my_plugin/images_similarity@v1"] 
+        name: str
+        # all properties apart from `type` and `name` are treated as either 
+        # definitions of batch-oriented data to be processed by block or its 
+        # parameters that influence execution of steps created based on block
+        image_1: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+            description="First image to calculate similarity",
+        )
+        image_2: Union[WorkflowImageSelector, StepOutputImageSelector] = Field(
+            description="Second image to calculate similarity",
+        )
+    ```
+    
+    * in the lines `2-9`, we've added a couple of imports ro ensure that we have everything needed
```

Comment:
`a couple of imports ro ensure` -> `a couple of imports to ensure`

Reply:
👍